core/src/graph/mod.rs

```diff
@@ -243,6 +243,27 @@ impl Type<MySql> for Value {
     }
 }
 
+macro_rules! encode_num_arrays {
+    ($n:expr, $arr:expr, $buf:expr; $($variant:ident),*) => {
+        match $n {
+            $(
+            Number::$variant(_) => {
+                let nums = $arr
+                    .iter()
+                    .map(|v| if let Some(Number::$variant(i)) = v.as_number() {
+                        *i
+                    } else { 
+                        panic!("inconsistent array number types")
```

Comment:
Won't that introduce a panic in a situation like an array of optional values?

Which raises another question: what are the semantics for failing an `encode_by_ref`? Or is the current API requiring `Value` to always be "encodable"?

Reply:
That's a good point. One of our big problems here is that the relation via `Encode` is not transitive (e.g. if something `encode`s into something that is `Encode`able, the latter won't still be `encode`d. We'll need a sqlx change to fix this, which will then also deal with nulls in integer arrays.